backend/czi_hosted/default_config.py

```diff
@@ -122,6 +122,15 @@
       # If timelimit_s is set to None, then there is no time limit.
       timelimit_s: 30
 
+    metadata_cache:
+      # The maximum number of datasets that may be opened at one time.  The least recently used dataset
+      # is evicted from the cache first.
+      max_datasets: 10000
+
+      # A matrix is automatically removed from the cache after timelimit_s number of seconds.
+      # If timelimit_s is set to None, then there is no time limit.
+      timelimit_s: 300
```

Comment:
Why is the time limit only 5 minutes? Seems like you would want longer?

Reply:
I was thinking that 5 minutes is long enough that loading a new dataset will only call the data portal once, but short enough that if we publish a revision the dataset location will be updated with minimal potential down time. 

But its super configurable/I'm open to changing it if you think a longer time period would be better. We could potentially add a delay longer than the cache length to the s3 object deletion when revising a dataset to avoid issues with downtime, but I'm a little worried about the user experience if a curator publishes a revision to a dataset and doesnt realize it will take a few minutes to propagate . What do you think?

Reply:
Your logic makes sense to me! Happy to try it out and configure it based on the experience.